File: proton-sdk/incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/UploadError.kt
package me.proton.drive.sdk.telemetry

enum class UploadError {
    UNRECOGNIZED,
    SERVER_ERROR,
    NETWORK_ERROR,
    INTEGRITY_ERROR,
    RATE_LIMITED,
    VALIDATION_ERROR,
    HTTP_CLIENT_SIDE_ERROR,
    UNKNOWN,
}

File: proton-sdk/client/cs/src/Proton.Drive.Sdk/Telemetry/UploadError.cs
namespace Proton.Drive.Sdk.Telemetry;

// Numeric values must match the UploadError enum in proton.drive.sdk.proto.
public enum UploadError
{
    ServerError = 0,
    NetworkError = 1,
    IntegrityError = 2,
    RateLimited = 3,
    HttpClientSideError = 4,
    Unknown = 5,
    ValidationError = 6,
}
